src/main/java/Bob.java

```diff
@@ -0,0 +1,207 @@
+                if (argument.isBlank()) {
+                    throw new MissingArgumentException("description of the todo");
+                }
+                Task task = new Todo(argument.strip());
+                list.add(task);
+                say("added: " + task);
+            }
+        },
+        DEADLINE("deadline") {
+            @Override
+            public void run() {
+                int byIndex = argument.lastIndexOf("/by ");
+                if (byIndex == -1) {
+                    throw new MissingArgumentException("'by' argument to add a deadline");
+                }
+                String desc = argument.substring(0, byIndex).strip();
+                if (desc.isBlank()) {
+                    throw new MissingArgumentException("description of the deadline");
+                }
+                Task task = new Deadline(
+                        desc,
+                        argument.substring(byIndex + 4).strip()
+                );
+                list.add(task);
+                say("added: " + task);
+            }
+        },
+        EVENT("event") {
+            @Override
+            public void run() {
+                int fromIndex = argument.lastIndexOf("/from ");
+                int toIndex = argument.lastIndexOf("/to ");
+                if (fromIndex == -1 || toIndex == -1) {
+                    throw new MissingArgumentException("'from' and 'to' arguments to add an event");
+                } else {
+                    Task task = getTask(fromIndex, toIndex);
+                    list.add(task);
+                    say("added: " + task);
+                }
+            }
+
+            private static Task getTask(int fromIndex, int toIndex) {
+                String desc = fromIndex < toIndex
+                                ? argument.substring(0, fromIndex)
+                                : argument.substring(0, toIndex);
+                desc = desc.strip();
+                String from = fromIndex < toIndex
+                                ? argument.substring(fromIndex + 6, toIndex)
+                                : argument.substring(fromIndex + 6);
+                from = from.strip();
+                String to = fromIndex < toIndex
+                                ? argument.substring(toIndex + 4)
+                                : argument.substring(toIndex + 4, fromIndex);
+                to = to.strip();
+                if (desc.isBlank()) {
+                    throw new MissingArgumentException("description of the event");
+                }
+                return new Event(desc, from, to);
+            }
+        },
+        MARK("mark") {
+            @Override
+            public void run() {
+                if (argument.isBlank()) {
+                    throw new MissingArgumentException("index of the task that you want to mark");
+                }
+                int i;
+                try {
+                    i = Integer.parseInt(argument) - 1;
```

Comment:
Not compulsory but could try to name i as something else. This can help other programmers understand the code easily.